legion/engine/physics/physx/systems/physx_physics_system.hpp

```diff
@@ -31,5 +31,7 @@ namespace legion::physics
         static constexpr float m_timeStep = 0.02f;
 
         physx::PxScene* m_physxScene;
+
+        std::mutex m_setupShutdownMutex;
```

Comment:
If this lock is not static between all instances of the system then it doesn't do anything XD

Reply:
also using async::spinlock and async::lock_guard will be a little nicer and technically a bit faster, although i doubt that matters much since it's startup and shutdown.
```
{
    async::lock_guard guard(PS::setupShutdownLock);
    // Do stuff
}
```